packages/flutter_tools/lib/src/base/process.dart

```diff
@@ -27,6 +27,7 @@ Future<int> runCommandAndStreamOutput(List<String> cmd, {
       // TODO(ianh): This doesn't handle IO buffering (where the data might be split half-way through a line)
       dataLines = dataLines.where((String s) => filter.hasMatch(s)).toList();
     }
+    dataLines = dataLines.where((String s) => s.isNotEmpty).toList();
```

Comment:
this is still going to be buggy. See the comment higher up. We should be properly buffering the text using a stream and a LineSplitter or something.